exchanges/coinbasepro/coinbasepro_test.go

```diff
@@ -2,1042 +2,2062 @@ package coinbasepro
+	resp, err := c.EditOrder(context.Background(), ordIDs.Orders[0].OrderID, testAmount, testPrice*10)
+	require.NoError(t, err)
+	assert.NotEmpty(t, resp, errExpectedNonEmpty)
+}
 
-	// CryptocurrencyWithdrawalFee Basic
-	feeBuilder = setFeeBuilder()
-	feeBuilder.FeeType = exchange.CryptocurrencyWithdrawalFee
-	if _, err := c.GetFee(context.Background(), feeBuilder); err != nil {
-		t.Error(err)
-	}
+func TestEditOrderPreview(t *testing.T) {
+	t.Parallel()
+	_, err := c.EditOrderPreview(context.Background(), "", 0, 0)
+	assert.ErrorIs(t, err, errOrderIDEmpty)
+	_, err = c.EditOrderPreview(context.Background(), "meow", 0, 0)
+	assert.ErrorIs(t, err, errSizeAndPriceZero)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c)
+	ordIDs := orderTestHelper(t, "")
+	resp, err := c.EditOrderPreview(context.Background(), ordIDs.Orders[0].OrderID, testAmount, testPrice*10)
+	require.NoError(t, err)
+	assert.NotEmpty(t, resp, errExpectedNonEmpty)
+}
 
-	// CryptocurrencyDepositFee Basic
-	feeBuilder = setFeeBuilder()
-	feeBuilder.FeeType = exchange.CryptocurrencyDepositFee
-	if _, err := c.GetFee(context.Background(), feeBuilder); err != nil {
-		t.Error(err)
-	}
+func TestGetAllOrders(t *testing.T) {
+	t.Parallel()
+	assets := []string{testFiat.String()}
+	status := make([]string, 2)
+	_, err := c.GetAllOrders(context.Background(), "", "", "", "", "", "", "", "", "", status, assets, 0, time.Unix(2, 2), time.Unix(1, 1))
+	assert.ErrorIs(t, err, common.ErrStartAfterEnd)
+	status[0] = "CANCELLED"
+	status[1] = "OPEN"
+	_, err = c.GetAllOrders(context.Background(), "", "", "", "", "", "", "", "", "", status, assets, 0, time.Time{}, time.Time{})
+	assert.ErrorIs(t, err, errOpenPairWithOtherTypes)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c)
+	status = make([]string, 0)
+	assets = make([]string, 1)
+	assets[0] = testCrypto.String()
+	_, err = c.GetAllOrders(context.Background(), "", testFiat.String(), "LIMIT", "SELL", "", "SPOT", "RETAIL_ADVANCED", "UNKNOWN_CONTRACT_EXPIRY_TYPE", "", status, assets, 10, time.Time{}, time.Time{})
+	assert.NoError(t, err)
+}
 
-	// InternationalBankDepositFee Basic
-	feeBuilder = setFeeBuilder()
-	feeBuilder.FeeType = exchange.InternationalBankDepositFee
-	feeBuilder.FiatCurrency = currency.EUR
-	if _, err := c.GetFee(context.Background(), feeBuilder); err != nil {
-		t.Error(err)
-	}
+func TestGetFills(t *testing.T) {
+	t.Parallel()
+	_, err := c.GetFills(context.Background(), "", "", "", time.Unix(2, 2), time.Unix(1, 1), 0)
+	assert.ErrorIs(t, err, common.ErrStartAfterEnd)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c)
+	_, err = c.GetFills(context.Background(), "", testPair.String(), "", time.Unix(1, 1), time.Now(), 5)
+	assert.NoError(t, err)
+	status := []string{"OPEN"}
+	ordID, err := c.GetAllOrders(context.Background(), "", "", "", "", "", "", "", "", "", status, nil, 3, time.Time{}, time.Time{})
+	assert.NoError(t, err)
+	if ordID == nil || len(ordID.Orders) == 0 {
+		t.Skip(skipInsufficientOrders)
+	}
```

Comment:
In future I wouldn't bother with this this handling, just adds more bloat. Devs can insert their own order IDs etc.